File: src/Learnify/Learnify.Core/Dto/Course/QuizQuestion/QuizQuestionResponse.cs
﻿using Learnify.Core.Dto.Attachment;
using Learnify.Core.Dto.Course.QuizQuestion.Answers;

namespace Learnify.Core.Dto.Course.QuizQuestion;

public class QuizQuestionResponse
{
    public string Id { get; set; }
    public string Question { get; set; }
    public AnswersResponse Answers { get; set; }
    public UserLessonQuizAnswerResponse UserAnswer { get; set; }
}

File: front/client/src/app/Models/Course/Lesson/QuizQuestion/QuizQuestionResponse.ts
import { AttachmentResponse } from "src/app/Models/Attachment/AttachmentResponse";
import { AnswersResponse } from "./Anwers/AnswersResponse";
import { UserLessonQuizAnswerResponse } from "./Anwers/UserLessonQuizAnswerResponse";

export interface QuizQuestionResponse {
    id: string;
    media: AttachmentResponse;
    question: string;
    answers: AnswersResponse;
    userAnswer: UserLessonQuizAnswerResponse;
}